### PR TITLE
Add i18n

### LINKED ISSUE #437: Add internationalization l18n support
We need to accommodate users of different languages!

Chromium has [a guide to adding templating to an extension here](https://www.chromium.org/developers/design-documents/extensions/how-the-extension-system-works/i18n).

Marking as bounty worthy. This bounty would not be for adding any particular translation, but to move all of our language from hard-coded within the application to template files so translations can be easily added.

Building the extension should still work on all supported browsers (Chrome, Firefox, Opera, Edge).

### PR DESCRIPTION
**Pull request by danfinlay**
Based on work here by @mapmeld:
https://github.com/MetaMask/metamask-extension/pull/3090

Fixes #437, I've confirmed it myself, although now that I made changes, it requires another review.

Also needs a merge conflict cleanup, I think one of the new UI devs would make the shortest work of this, either @danjm or @alextsg.

### CODE REVIEW

**Review by danjm — APPROVED** [review]
@danfinlay I resolved merged conflict, QA'd locally and reviewed code.

I ran into an issue last night where the translastions were not rendering, but it fixed itself when I removed the local extension and re-added it. @alextsg tested locally as well, and verified that it works.

Approved.

File: ui/i18n.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      phrase = phrase.replace(/\$1/g, substitutions[0])
+      if (substitutions.length > 1) {
+        phrase = phrase.replace(/\$2/g, substitutions[1])
+      }
+    }
+    return phrase
+  }
+}
+
+window.h = getMessage
```
Comment (mapmeld) [inline ui/i18n.js]: window.h ?

File: ui/app/components/signature-request.js [hunk 2]
```diff
@@ -54,7 +55,7 @@ SignatureRequest.prototype.renderHeader = function () {
 
     h('div.request-signature__header-background'),
 
-    h('div.request-signature__header__text', 'Signature Request'),
+    h('div.request-signature__header__text', t('sigRequest')),
```
Comment (tmashuang) [inline ui/app/components/signature-request.js]: I don't see `sigRequest` in `app/_locales/en`. I see `sigRequested` and `yourSigRequested`. @mapmeld Was `sigRequested` the preferred one?

Comment (mapmeld) [reply ui/app/components/signature-request.js]: It looks like there are actually three different messages used ("Signature Request", "Signature Requested", "Your Signature Requested").  I would recommend either adding a ```sigRequest``` entry to the locales file, or changing the line here (and message) to ```t('sigRequested')```

Comment (tmashuang) [reply ui/app/components/signature-request.js]: Ok thanks for the feedback. Also, is it a typo `sigRequest` vs `signRequest`?

Comment (mapmeld) [reply ui/app/components/signature-request.js]: No I don't think so... my guess is when I alphabetized the messages, I didn't notice that "Signature Request" and "Signature Requested" were different.
If you want the UI to be exactly the same, add ```sigRequest``` as an entry in the locales file.
If you want to change this one line, it has to be a code already present in the locales file (```sigRequested``` would work if you feel the message has the same meaning)

**Review by kumavis — APPROVED** [review]

**Comment by kumavis** [comment]
historic moment

**Pull request merged**